doc/tb3_tutorial/README.md

```diff
@@ -0,0 +1,573 @@
+
+    sudo rm -rf /etc/ros/rosdep/source.list.d/*
+
+**Install dependencies using rosdep**
+
+    rosdep install --from-paths src --ignore-src --rosdistro crystal -y --skip-keys "console_bridge fastcdr fastrtps libopensplice67 libopensplice69 rti-connext-dds-5.3.1 urdfdom_headers"
+    
+    python3 -m pip install -U lark-parser
+
+**Build ROS2**
+
+    colcon build --symlink-install
+
+or
+
+    colcon build --symlink-install --merge-install
+
+*Note*: I recommend you use --merge-install option since it will merge all the setup.bash files. Currently, sourcing takes a bit longer than usual, so this may save you some time when you open a new terminal.
+
+## Test ROS2
+
+We are done building ROS2. Now, it is time to test it!
+
+The easiest way to do it is using the ROS's famous talker and listener examples. Let's run the talker node first. What the talker does is publishing messages ("Hello World") to a topic. We will use the listener node to subscribe to the same topic to get these published messages and print them in the terminal.
+
+Open a new terminal and run (this node is written in C++)
+
+    source ~/ros2_all_ws/ros2_ws/install/local_setup.bash
+    ros2 run demo_nodes_cpp talker
+
+Open another terminal and run (this node is written in Python)
+
+    source ~/ros2_all_ws/ros2_ws/install/local_setup.bash
+    ros2 run demo_nodes_py listener
+   
+After running both nodes, you should see results similar to this.
+
+![enter image description here](https://github.com/mlherd/ros2/blob/master/img/ros2_test.PNG?raw=true)
+
+#  ROS2-Gazebo Installation
+
+We will be using Gazebo with ROS 2, so we need to install the Gazebo packages for ROS2. We will test ROS2-Gazebo after installing Gazebo 9. The fallowing command should install all the necessary ros-gazebo packages for ROS Dashing distribution.
+
+    sudo apt-get install ros-dashing-gazebo*
+
+# Gazebo Installation
+
+We will use Gazebo as our simulator, so let's install Gazebo 9. You can find more information about Gazebo [**here**](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&ved=2ahUKEwj_wKXW-vniAhXgJTQIHQmGA2EQFjAAegQIABAB&url=http://gazebosim.org/&usg=AOvVaw3bYv0Z_AA1EEyIXg8sGUuF). 
+
+    curl -sSL http://get.gazebosim.org | sh
+
+If for some reason this doesn't work, you can also use `sudo apt-get install gazebo9` to install Gazebo.
+
+If you see a message saying Gazebo is already installed you can skip this step. That means Gazebo is already installed.
+
+## Test Gazebo
+
+Let's test ROS2-Gazebo and Gazebo together.
+
+    sudo apt install ros-dashing-ros-core ros-dashing-geometry2
+    gazebo --verbose /opt/ros/dashing/share/gazebo_plugins/worlds/gazebo_ros_diff_drive_demo.world
+
+If Gazebo prints "plugin can't be found" error message, try running it again
+
+    soruce /opt/ros/dashing/setup.bash
```

Comment:
You built ROS 2 from source, but, instead of sourcing that workspace, you are sourcing the binary install directory. Which one do you want to use? It should be consistent through the tutorial IMHO.

Reply:
Fixed it. Now, I source the setup.bash file in ros2_ws workspace.